src/Security/Member.php

```diff
@@ -872,12 +872,50 @@ public static function currentUser()
     }
 
     /**
+     * Allow override of the current user ID
+     *
+     * @var int|null Set to null to fallback to session, or an explicit ID
+     */
+    protected static $overrideID = null;
+
+    /**
+     * Temporarily act as the specified user, limited to a $callback, but
+     * without logging in as that user.
+     *
+     * E.g.
+     * <code>
+     * Member::logInAs(Security::findAnAdministrator(), function() {
+     *     $record->write();
+     * });
+     * </code>
+     *
+     * @param Member|null|int $member Member or member ID to log in as.
+     * Set to null or 0 to act as a logged out user.
+     * @param $callback
+     */
+    public static function actAs($member, $callback)
+    {
+        $id = ($member instanceof Member ? $member->ID : $member) ?: 0;
+        $previousID = static::$overrideID;
+        static::$overrideID = $id;
+        try {
+            return $callback();
+        } finally {
```

Comment:
My only gripe here is that any associated error handling is executed in the current user context as well - project might've been configured to output stack traces based on user permissions, rather than environment context. But that's enough of an edge case to ignore it.